### PR TITLE
Implement Index a default file

### PR DESCRIPTION
**Pull request by john681611**
Attempting to fix https://github.com/sass/sass/issues/690

Simple implementation that if no file in current level is found it looks if it's a directory and searches for a valid index file inside.

Though I have the tests working I am unsure of how to run the code locally in a manual test.
 
Working with:
@liamshaunsmith
@mariann013 

See sass/sass-spec#1206

### CODE REVIEW

**Comment by nex3** [comment]
Thanks for the contribution!

Because we're in the process of [deprecating Ruby Sass](http://sass.logdown.com/posts/1022316-announcing-dart-sass), we aren't currently adding new features to Ruby only. If you're willing to add this behavior to [Dart Sass](https://github.com/sass/dart-sass) as well and add cross-platform tests to [sass-spec](https://github.com/sass/sass-spec), then we can land it in both places at once.

**Comment by john681611** [comment]
Ok I've got zero experience in dart so I will have a go

**Comment by nex3** [comment]
Give it a shot! Dart is pretty easy to learn.

**Comment by nex3** [comment]
Same comments as on the Dart PR: if the import ends in `.scss` or `.sass` it shouldn't look for the index, and the tests aren't necessary here.

File: lib/sass/importers/filesystem.rb [hunk 1]
```diff
@@ -154,7 +154,9 @@ def find_real_file(dir, name, options)
             [Sass::Util.cleanpath(full_path).to_s, s]
           end
         end.flatten(1)
-        return if found.empty?
+        if found.empty? && _extension?(name) && File.directory?("#{dir}/#{name}")
```
Comment (nex3) [inline lib/sass/importers/filesystem.rb]: I don't understand what `_extension?(name)` is doing here... are we only checking for an index file if `name` has an extension? That seems wrong.

Comment (john681611) [reply lib/sass/importers/filesystem.rb]: We figured it would be more readable than `split(name)[2].nil?` I'll switch back

Comment (nex3) [reply lib/sass/importers/filesystem.rb]: I meant more "why do we need to check for an extension at all" than "why is it factored out into a private method"

Comment (john681611) [reply lib/sass/importers/filesystem.rb]: Because of your above comment that if it has an extension in the name we shouldn't look at all.

Comment (nex3) [reply lib/sass/importers/filesystem.rb]: I just meant a `.sass` or `.scss` extension. If we disallow *any* extension, [this spec](https://github.com/john681611/sass-spec/tree/bd1e7759b7641cc88b4f1a49b7058e08e5a48b42/spec/directives/import/index/dir_dot_foo) will fail (and in fact I'm not sure why it isn't failing already...)

Comment (nex3) [reply lib/sass/importers/filesystem.rb]: Oh, sorry, I was being dumb here... I didn't realize that this was using the `split()` method defined below that already checks for the correct file extensions. Your original code was fine; I'll revert b196d9c.

**Comment by AndrewBogdanovTSS** [comment]
Any info on when this feature might be released?

**Comment by john681611** [comment]
I believe its released in https://sass-lang.com/dart-sass, ruby-sass is being deprecated so I'd look into moving to that.

**Comment by nex3** [comment]
This was released as part of Ruby Sass 3.6.0.

**Comment by john681611** [comment]
> This was released as part of Ruby Sass 3.6.0.

Something I have been wondering is did this make it into lib-sass or is that another separate piece of work I need to do?

Basically, we need a web pack sass-loader that supports dart-sass and does de-duping.

**Comment by nex3** [comment]
> Something I have been wondering is did this make it into lib-sass or is that another separate piece of work I need to do?

It looks like this landed in https://github.com/sass/libsass/pull/2614.

> Basically, we need a web pack sass-loader that supports dart-sass and does de-duping.

The Webpack Sass loader support Dart Sass now. You just need to pass `implementation: require("sass")` to it.

**Pull request merged**